lang/values.md

```diff
@@ -185,27 +185,24 @@ Note that types like `&!` have no valid value: when the pointee type is uninhabi
   (The latter would not even be possible with the current structure of MiniRust.)
   Also see [this discussion](https://github.com/rust-lang/unsafe-code-guidelines/issues/77).
 
-### Tuples (and arrays, structs, ...)
-
-For simplicity, we only define pairs for now.
+### Tuples (and structs, ...)
 
 ```rust
 impl Type {
-    fn decode(Tuple { fields: [field1, field2], size }: Self, bytes: List<AbstractByte>) -> Option<Value> {
+    fn decode(Tuple { fields, size }: Self, bytes: List<AbstractByte>) -> Option<Value> {
         if bytes.len() != size { throw!(); }
-        let (offset1, type1) = field1;
-        let val1 = type1.decode(bytes[offset1..][..type1.size()]);
-        let (offset2, type2) = field2;
-        let val2 = type2.decode(bytes[offset2..][..type2.size()]);
-        Value::Tuple([val1, val2])
+        Value::Tuple(
+            fields.into_iter().map(|(offset, ty)| {
+                ty.decode(bytes[offset..][..ty.size()])
+            }).try_collect()?,
```

Comment:
Can't `collect` also do this?

Reply:
Yeah for `Result` `collect` also works, so given that `try_collect` is unstable IMO we should use `collect` for now.